### ISSUE TITLE
OSGi Manifest: optional Ant dependencies in OSGi metadata

### ISSUE THREAD

**Issue by jjoslet** [janino-compiler/janino#86]
The dependencies to ANT is only used when running Janino as an Ant compiler (http://janino-compiler.github.io/janino/#janino-as-an-ant-compiler). Therefore, this dependency should be marked optional in the OSGi metadata since it is not needed for a common uses.

**Comment by aunkrig** [comment]
Hi,

it has scope "provided"; isn't that correct?

**Comment by jjoslet** [comment]
I think it should be optional too.

From the Maven documentation (https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html):

> This is much like compile, but indicates you expect the JDK or a container to provide the dependency at runtime. For example, when building a web application for the Java Enterprise Edition, you would set the dependency on the Servlet API and related Java EE APIs to scope provided because the web container provides those classes. This scope is only available on the compilation and test classpath, and is not transitive.

Without optional, this means that every container or JDK running Janino should provide the Ant dependencies. In your case, you just need it when running inside Ant, which is an optional way to work.

This currenlty works with a standard Java application since there is no dependency verification at runtime. With OSGi, this is not the case; the OSGi framework validates that all dependencies required by the bundle (the jar) is present. If you require Ant (provided means that this is required at runtime to work), the framework complains if there is no Ant in the application.

If the Ant import is optional for OSGi, the framework will no more complains if Ant is not present. But if Ant is present, it'll give the dependency to the JAR, as expected.

Writing the dependency like this:
```
<dependency>
    <groupId>org.apache.ant</groupId>
    <artifactId>ant</artifactId>
    <version>1.7.1</version>
    <scope>provided</scope>
    <optional>true</optional>
</dependency>
```

will not impact standard Maven application but the `maven-bundle-plugin` will interpret the `optional` parameter to mark the dependency as optional in the `MANIFEST.MF` file for OSGi.

**Comment by aunkrig** [comment]
Just added the "<optional>" line. Please test.
Do you want me to prepare a new release?

**Comment by jjoslet** [comment]
I've tested your commit and it solves the problem.

There is no need to provide a new release only for that, this is not blocking for me, you can wait for more content.

Thanks.

**Comment by aunkrig** [comment]
👍

**Comment by aunkrig** [comment]
FYI: I just released version 3.0.13, so the fix is now available in an official release.
